mlir/lib/Dialect/MIOpen/Tuning/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 add_mlir_dialect_library(MLIRMIOpenTuning
   SqliteDb.cpp
   GridwiseGemmParams.cpp
+  BackwardDataPaddingKernel.cpp
```

Comment:
@krzysz00 this file should not be a part of MLIRMIOpenTuning, it's part of the lowering process. If you take a look at my other PR, #293 , which also tries to address some of the issues caused by PR #279 , I kept the file still as a header file, and mark them `inline`.

Reply:
You didn't actually commit that change.

And while it maybe shouldn't be part of tuning, I still don't think we should have that much code in a header file without a good argument for it (that is, it's got templates or the like).

Do you have a better suggestion for where to put it?

(I put this in tuning since, from what I could tell, the only users of this were tuning people, but I might've missed something)

Reply:
@krzysz00 my bad, it's #292 you need to check, NOT #293.

Reply:
Yeah, true

I still feel like that's bad style and would like to move those functions into a `.cpp`

Reply:
As for the big header file issue, let's tackle them in a (or more) separate PRs. Let's focus on getting rid of compiler errors caused by #279 here. On my node with `g++ 7.5.0` they are just warnings, but it seems it's causing failures in other configs.

Reply:
Fair enough. And yeah, it's a hard error on various `clang` versions, possibly as a result of the c++ standard in use?

Since it looks like I've accidentally duplicated your work (this is what I get for not reading my email properly, but maybe this is also a function of remote being hard), I'll close this and we can move to yours.

Reply:
For the proper library to host the logic converted from the huge header to separated `.cpp` , I think `MLIRMIOpenTransforms` should be a better target.